.github/workflows/e2e.yaml

```diff
@@ -40,6 +40,12 @@ jobs:
           echo "Building collector artifact"
           make build
 
+      - name: Upload binary artifact
```

Comment:
Did you mean to make these changes in the same PR?

Reply:
This is step necessary to grab the binary later on when we run the E2E tests for the provider. The rest of the changes are just to run the tests.

Reply:
update: i see it's required later in e2e tests